src/context/AuthContext.tsx

```diff
@@ -35,18 +35,20 @@ export const AuthProvider: ChildrenFC = ({ children }) => {
   const [user, setUser] = useState(defaultUser);
 
   useEffectAsync(async () => {
-    const { token } = nookies.get();
-
-    if (!token) {
-      return;
-    }
-
-    const userData = await getUser(() => {
+    try {
+      const { token } = nookies.get();
+
+      if (!token) {
+        return;
+      }
+
+      const userData = await getUser();
+      if (userData) {
+        setUser(userData);
+      }
+    } catch (error) {
       nookies.destroy(null, 'token');
-    });
-
-    if (userData) {
-      setUser(userData);
+      setUser(defaultUser);
```

Comment:
I dont think it should be named defaultUser - even I might have suggested it. Let's call it anonymousUser, emptyUser or something like that